Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.Expression.cs

```diff
@@ -5566,8 +5566,14 @@ public sealed class Expression<TDelegate>
   {
     private Expression() { }
 
-    // public TDelegate Compile();
+#if !SILVERLIGHT
+        public TDelegate Compile()
+        {
+            Contract.Ensures(!ReferenceEquals(Contract.Result<TDelegate>(), null));

```

Comment:
Probably right but I am curious, why not `Contract.Result<TDelegate>() != null` ?